python/metatensor-learn/metatensor/learn/data/dataset.py

```diff
@@ -0,0 +1,126 @@
+"""
+Module for defining a dataset class.
+"""
+import os
+from typing import List, Union, Dict, Optional
+import torch
+
+
+class Dataset(torch.utils.data.Dataset):
+    """
+    Class for defining a torch dataset.
+
+    `data` must be a dict of data items indexed by their str name. Each data
+    item is either a list or a dict.
+    """
+
+    def __init__(
+        self,
+        data: Dict[str, Union[List, Dict]],
```

Comment:
Sorry for forgetting, what was the use case of using a `Dict[str, Dict]` again? If you want to give your samples more meta information, like structure, so we are agnostic to the data format used (TensorMap atomistic.System)?

Reply:
To be fair, I think that there isn't a use case for this. I have removed this, and required that data is passed as `Dict[str, List]`. A non-continuous range of indices can be passed in `sample_idxs`, that will be zipped 1:1 with the elements in the data items